_posts/2018-10-17-using-environment-variables-with-jekyll-vue.md

```diff
@@ -0,0 +1,103 @@
+---
+layout: post
+title: "Using Environment Variables with Jekyll, Vue.js, and Netlify"
+date: 2018-10-17 07:17:33
+tags:
+- Development
+- JavaScript
+- Dev Ops
+- Vue
+- Vue.js
+- Jekyll
+categories: development
+twitter_text: "Using Environment Variables with Jekyll, Vue.js, and Netlify"
+authors: John Vicari
+image: https://images.unsplash.com/photo-1489417139533-915815598d31?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=2642373de2eddf3629e9ff52550c8294&auto=format&fit=crop&w=2820&q=80
+image_url: https://unsplash.com/photos/9FDKj-FrfA4
+image_credit: Photo by Rebekah Howell on Unsplash
+---
+
+We operate in a typical `QA -> Production` workflow on both our front-end and back-end teams. So, what happnes when your front-end is in QA but is pointing to a Production API? Nothing good happens, son... nothing good.
+
+As I've [mentioned before](/ie-still-breaking-promises-literally/), we run a few websites built with [Jekyll](https://jekyllrb.com/) and [Vue.js](https://vuejs.org/), and recently launched a new site with a similar setup. Since this site had never been released, we hardcoded a handful of QA endpoints into a Vue component. This didn't present any issues during initial development, but when it was released (and the endpoints were updated to Production) and minor bugs were being worked on, front-end `QA` was pointing to back-end `Production`. Not good, especially if data is being manipulated.
```

Comment:
Using `post_url` can help validate the link is correct.

```suggestion
As I've [mentioned before]({% post_url 2018-06-08-ie-still-breaking-promises-literally %}), we run a few websites built with [Jekyll](https://jekyllrb.com/) and [Vue.js](https://vuejs.org/), and recently launched a new site with a similar setup. Since this site had never been released, we hardcoded a handful of QA endpoints into a Vue component. This didn't present any issues during initial development, but when it was released (and the endpoints were updated to Production) and minor bugs were being worked on, front-end `QA` was pointing to back-end `Production`. Not good, especially if data is being manipulated.
```